src/r_cmd.rs

```diff
@@ -63,58 +63,97 @@ pub struct RCommandLine {
     pub r: Option<PathBuf>,
 }
 
-pub fn find_r_version_command(r_version: &Version) -> Option<RCommandLine> {
-    // Give preference to the R version on the $PATH
-    if does_r_cmd_match_version(&RCommandLine { r: None }, r_version) {
-        log::debug!("R {r_version} found on the path");
-        return Some(RCommandLine { r: None });
+pub fn find_r_version_command(r_version: &Version) -> Result<RCommandLine, VersionError> {
+    let mut found_r_vers = Vec::new();
+    // Give preference to the R version on the path
+    if let Ok(path_r) = (&RCommandLine { r: None }).version() {
+        if r_version.hazy_match(&path_r) {
+            log::debug!("R {r_version} found on the path");
+            return Ok(RCommandLine { r: None });
+        }
+        found_r_vers.push(path_r);
     }
 
     // For windows, R installed/managed by rig is has the extension .bat
-    if cfg!(windows)
-        && does_r_cmd_match_version(
-            &RCommandLine {
-                r: Some(PathBuf::from("R.bat")),
-            },
-            r_version,
-        )
-    {
-        log::debug!("R {r_version} found on the path from `rig`");
-        return Some(RCommandLine {
+    if cfg!(windows) {
+        if let Ok(rig_r) = (&RCommandLine {
             r: Some(PathBuf::from("R.bat")),
-        });
+        })
+            .version()
+        {
+            if r_version.hazy_match(&rig_r) {
+                log::debug!("R {r_version} found on the path from `rig`");
+                return Ok(RCommandLine {
+                    r: Some(PathBuf::from("R.bat")),
+                });
+            }
+            found_r_vers.push(rig_r);
+        }
     }
 
     let opt_r = PathBuf::from("/opt/R");
     if !opt_r.is_dir() {
-        return None;
+        if found_r_vers.is_empty() {
```

Comment:
No need to duplicate that whole thing. Just put the iteration on opt/R in the `if opt_r.is_dir()`